### PR TITLE
Fixup nox

### PR DESCRIPTION
**Pull request by jonparrott**
This enables grpc tests on python 3 and unifies the app engine and python 2.7 test suite (woohoo).

### CODE REVIEW

File: bigtable/hello/main.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def main(project_id, instance_id, table_id):
             row = table.row(row_key)
             row.set_cell(
                 column_family_id,
-                column_id.encode('utf-8'),
+                column_id,
```
Comment (jerjou) [inline bigtable/hello/main.py]: Should this have some `six` magic for python2 compatibility?

**Comment by waprin** [comment]
hooray. code lgtm. running it locally if you want to wait for that but I'm guessing any problems there will be doc issues anyway.

File: bigtable/hello/main.py (hunk 1 above)
Comment (waprin) [inline bigtable/hello/main.py]: I think Jon is just fixing a double encode? I think in general if you want Python strings as bytes, do the encode, no six necessary.

File: appengine/standard/conftest.py [hunk 2]
```diff
@@ -25,3 +30,13 @@
 (pytest_runtest_call)
 (run_tasks)
 (testbed)
+
+
+def pytest_ignore_collect(path, config):
+    """Skip App Engine tests in python 3 and if no SDK is available."""
```
Comment (jerjou) [inline appengine/standard/conftest.py]: "and" -> "or"

File: nox.py [hunk 3]
```diff
@@ -132,8 +121,11 @@ def filter_samples(sample_dirs, changed_files):
 def setup_appengine(session):
     """Installs the App Engine SDK."""
     # Install the app engine sdk and setup import paths.
+    if session.interpreter.startswith('python3'):
+        return
```
Comment (jerjou) [inline nox.py]: I feel like this check makes more sense in the caller, no?

File: bigtable/hello/main.py (hunk 1 above)
Comment (jerjou) [inline bigtable/hello/main.py]: Ah cool. I totally didn't look at the context ^_^;

**Comment by jerjou** [comment]
LGTM if LGTTravis

File: bigtable/hello/main.py (hunk 1 above)
Comment (jonparrott) [inline bigtable/hello/main.py]: Yeah fixing a double encode. No idea how this worked on py2.7 other than luck.

File: appengine/standard/conftest.py (hunk 2 above)
Comment (jonparrott) [inline appengine/standard/conftest.py]: Done.

File: nox.py (hunk 3 above)
Comment (jonparrott) [inline nox.py]: meh.

**Comment by jonparrott** [comment]
@dpebot merge when travis passes

**Pull request merged**